src/apply.jl

```diff
@@ -25,16 +25,49 @@ function ITensors.apply(
     if ortho
       ψ = orthogonalize(ψ, v⃗[1])
     end
-    oψᵥ = apply(o, ψ[v⃗[1]] * ψ[v⃗[2]])
-    ψᵥ₁, ψᵥ₂ = factorize(
-      oψᵥ, inds(ψ[v⃗[1]]); cutoff, maxdim, tags=ITensorNetworks.edge_tag(e)
+
+    #Check whether to do a memory efficient QR first (do both sites combined for now, although we could split if we wish)
+    #if no maxdim is provided just do it based on inds counting, if it is provided do it based on whichever takes less memory
+
+    v1_siteinds = setdiff(
+      inds(ψ[v⃗[1]]), flatten(unique(inds.([ψ[v] for v in neighbors(ψ, v⃗[1])])))
     )
+    v2_siteinds = setdiff(
+      inds(ψ[v⃗[2]]), flatten(unique(inds.([ψ[v] for v in neighbors(ψ, v⃗[2])])))
+    )
+    d1, d2 = prod(dim.(v1_siteinds)), prod(dim.(v2_siteinds))
+    χv1, χv2 = prod(
+      dim.(uniqueinds(uniqueinds(ψ[v⃗[1]], ψ[v⃗[2]]), inds(ψ[v⃗[1]]; tags="Site")))
+    ),
+    prod(dim.(uniqueinds(uniqueinds(ψ[v⃗[2]], ψ[v⃗[1]]), inds(ψ[v⃗[2]]; tags="Site"))))
+    if (maxdim == nothing && length(inds(ψ[v⃗[1]])) + length(inds(ψ[v⃗[2]])) <= 6) ||
+      (maxdim != nothing && maxdim * maxdim * d1 * d2 >= χv1 * χv2)
+      oψᵥ = apply(o, ψ[v⃗[1]] * ψ[v⃗[2]])
+      ψᵥ₁, ψᵥ₂ = factorize(
+        oψᵥ, inds(ψ[v⃗[1]]); cutoff, maxdim, tags=ITensorNetworks.edge_tag(e)
+      )
+    else
+      Qv1, Rv1 = factorize(
+        ψ[v⃗[1]], uniqueinds(uniqueinds(ψ[v⃗[1]], ψ[v⃗[2]]), inds(ψ[v⃗[1]]; tags="Site"))
```

Comment:
Again, I would hope we could avoid using `tags` to filter indices. I thought `uniqueinds(ψ[v⃗[1]], ψ[v⃗[2]])` would be sufficient here?

Reply:
But you don't want the "Site" index to go onto `Qv1` no? Which is what would happen if you just did `uniqueinds(ψ[v⃗[1]], ψ[v⃗[2]])`

Reply:
I.e. you need `uniqueinds(uniqueinds(ψ[v⃗[1]], ψ[v⃗[2]]), uniqueinds(ψ, v⃗[1]))` (if you want to avoid tag specification - which makes sense).

Reply:
Right I forgot you want the site indices to go onto `Rv1`.